modeshape-jcr/src/main/java/org/modeshape/jcr/journal/LocalJournal.java

```diff
@@ -265,6 +258,19 @@ public Records recordsNewerThan( DateTime changeSetTime,
     }
 
     @Override
+    public Set<NodeKey> changedNodesSince( long timestamp ) {
+        Records records = recordsNewerThan(new DateTime(timestamp), true, false);
+        if (records.isEmpty()) {
+            return Collections.emptySet();
+        }
+        Set<NodeKey> results = new LinkedHashSet<>();
+        for (JournalRecord record : records) {
+            results.addAll(record.getChangeSet().changedNodes());
+        }
+        return results;
```

Comment:
This seems like it'd be pretty straightforward to return `Iterable<NodeKey>` here that would return a custom `Iterator` that would walk the records during `hasNext()` and/or `next()`.